src/filehandler/pemhandler.cpp

```diff
@@ -1,40 +1,50 @@
 #include "filehandler/pemhandler.h"
+#include "params.h"
 #include "base64.h"
 #include "hex.h"
 
 #include <fstream>
 #include <iostream>
 #include <sodium.h>
+#include <stdexcept>
+#include <iomanip>
 
 namespace ih
 {
-PemFileReader::PemFileReader(std::string const filePath) :
+PemFileReader::PemFileReader(std::string const &filePath) :
         IFileHandler(filePath)
 {
-    m_fileContent = getFileContent();
+    if (PemFileReader::isFileValid())
```

Comment:
Why not `throw` when the file is not valid?

Reply:
Because the thrown exception was going to be propagated upwards to the caller of the constructor anyway.
But you are right, it's better to actually to use a try-catch-throw block in the constructor. I've changed it